feat(csp): document report-only usage and usecase with adyen (#496)
front-commerce/developers.front-commerce.com: docs/reference/content-security-policy.mdx

---
title: Content Security Policy
description: This guide is about Content Security Policies configuration.
---

This guide explains how to setup
<abbr title="Content Security Policy">CSP</abbr> in your [config/website.js
configuration](/docs/reference/configurations#configwebsitejs).

## Basic configuration

For security reasons only URLs from the store's domain are authorized through

> CSP. However, for tracking and external dependencies, we may authorize more

Reviewer: ```suggestion
CSP. However, for tracking and third-party dependencies, you will have to authorize more
```

domains. Use the following config and add your custom domain in each:

```js title="my-module/config/website.js"
{
  ...
  // highlight-start
  contentSecurityPolicy: {
    directives: {
      scriptSrc: [],
      frameSrc: [],
      styleSrc: [],
      imgSrc: [],
      fontSrc: [],
      connectSrc: [],
      baseUri: []
    }
  }
  // highlight-end
}
```

When a CSP is missing, your browser will let you know in its console and log it

> using the `security` logger

Reviewer: ```suggestion
using the `security` logger.
```

## Enable Report-Only

:::warning Loose security policies

This chapter implies lowering the security of your application, please fully

> consider your needs before using this feature

Reviewer: ```suggestion
consider your needs before using this feature.
```
